### PR TITLE
Prepare 0.4.0 release

### PR DESCRIPTION
**Pull request by nikola-jokic**
Notable changes (Excluding tests):
- #2382
- #2276
- #2363
- #2374
- #2391
- #2450
- #2435
- #2418
- #2433
- #2477
- #2480

### CODE REVIEW

File: docs/preview/gha-runner-scale-set-controller/README.md [hunk 1]
```diff
@@ -219,6 +219,25 @@ To fix this, you can either:
 
 ## Changelog
 
+### v0.4.0
+
+#### Major changes
+
```
Comment (Link-) [inline docs/preview/gha-runner-scale-set-controller/README.md:250]: I think we should add a warning here that this release contains breaking changes and that end-users are advised to evaluate carefully before upgrading

File: docs/preview/gha-runner-scale-set-controller/README.md [hunk 2]
```diff
@@ -219,6 +219,25 @@ To fix this, you can either:
...
+1. Surface EphemeralRunnerSet stats to AutoscalingRunnerSet [#2382](https://github.com/actions/actions-runner-controller/pull/2382)
+1. Remove list/watch secrets permission from manager cluster role
```
Comment (Link-) [inline docs/preview/gha-runner-scale-set-controller/README.md]: I think we should have instructions for this upgrade i.e. how folks should go about it without breaking their environments

File: docs/preview/gha-runner-scale-set-controller/README.md (hunk 1 above)
Comment (nikola-jokic) [reply docs/preview/gha-runner-scale-set-controller/README.md:250]: Added warning section. Let me know if I should phrase it differently :relaxed:

**Comment by Link-** [comment]
### TODO

- [ ] After merge, create and push tag: `gha-runner-scale-set-0.4.0`
- [ ] Verify new chart has been pushed to: https://github.com/actions/actions-runner-controller/pkgs/container/actions-runner-controller-charts%2Fgha-runner-scale-set-controller
- [ ] Verify new chart has been pushed to: https://github.com/actions/actions-runner-controller/pkgs/container/actions-runner-controller-charts%2Fgha-runner-scale-set
- [ ] Verify new image has been pushed to: https://github.com/actions/actions-runner-controller/pkgs/container/gha-runner-scale-set-controller

**Review by Link- — APPROVED** [review]

**Comment by Edgarruiz8585** [comment]
> Notable changes (Excluding tests):
> - #2382
> - #2276
> - #2363
> - #2374
> - #2391
> - #2450
> - #2435
> - #2418
> - #2433
> - #2477
> - #2480

- [x] by Edgarruiz8585

File: docs/preview/gha-runner-scale-set-controller/README.md (hunk 1 above)
Comment (Edgarruiz8585) [reply docs/preview/gha-runner-scale-set-controller/README.md:250]: By Edgarruiz8585

**Pull request merged**